txselector/txselector.go

```diff
@@ -144,6 +144,12 @@ func (txsel *TxSelector) GetL1L2TxSelection(selectionConfig txprocessor.Config,
 		discardedL2Txs, tracerr.Wrap(err)
 }
 
+type failedAtomicGroup struct {
```

Comment:
since this struct is meant to contain only one transaction from the atomic group, I suggest calling this as `failedAtomicTransaction`

Reply:
The `txID` is used to indicate the tx that made the entire group of txs fail. But the names I choose are not self explainatory, so I'll update this